calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -198,13 +198,17 @@ func (rh resourceHelper) Update(ctx context.Context, client client.Interface, re
 	// If the resourceVersion is specified then we use it to try and update the resource.
 	// Do not attempt to retry if the resource version is specified.
 	if rv != "" {
+		// Clean out the resource version to always get the latest revision.
+		resource.(ResourceObject).GetObjectMeta().SetResourceVersion("")
```

Comment:
This change means that the method now modified the input data.  I feel pretty strongly that we shouldn't be modifying what's passed in.  It makes for some pretty sneaky bugs down the line and the signature of this method makes it look like it shouldn't touch its input (because it _returns_ a copy of the object with the new values filled in).  

For an example of where the bugs would pop out: a similar issue tripped me up a couple of days ago: my canned unit test data was being modified when I passed it into libcalico-go so the first test that used the canned data worked but later tests weren't sending in the data that they expected.

I think the answer is simply to take a copy of the input struct if you need to modify it.  I think there's a DeepCopy() method on the object that you could use for that.

Reply:
Good point, I overlooked the fact that it will affect our input values.